src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -247,6 +250,52 @@ public bool TryParseEofHeader(ReadOnlySpan<byte> container, [NotNullWhen(true)]
 
         bool ValidateBody(ReadOnlySpan<byte> container, ref EofHeader? header)
         {
+            SectionHeader[]? codeSections = header.Value.CodeSections;
+            (int typeSectionStart, ushort typeSectionSize) = header.Value.TypeSection;
+            if (codeSections.Length == 0 || codeSections.Any(section => section.Size == 0))
+            {
+                if (_loggingEnabled)
+                {
+                    _logger.Trace($"EIP-3540 : CodeSection size must follow a CodeSection, CodeSection length was {codeSections.Length}");
+                }
+
+                header = null;
+                return false;
+            }
+
+            if (codeSections.Length != (typeSectionSize / 4))
```

Comment:
what is for 4? avoid magic numbers

Reply:
4 is the minimum allowed size for typesections, which makes it the size a function typedata takes in the typesection (1 bytes for input count, 1 byte for output count, 2 bytes for max data stack height), I moved it to const field as well but I think MINIMUM_TYPESECTION_SIZE doesn't reflect what it does.